surefire-providers/surefire-testng-utils/src/main/java/org/apache/maven/surefire/testng/utils/MethodSelector.java

```diff
@@ -62,6 +62,17 @@ private static boolean shouldRun( ITestNGMethod test )
     {
         TestListResolver resolver = testListResolver;
         boolean hasTestResolver = resolver != null && !resolver.isEmpty();
-        return hasTestResolver && resolver.shouldRun( test.getRealClass(), test.getMethodName() );
+        boolean resolved = false;
+        Class<?> clazz = test.getRealClass();
+        if (resolver != null) {
```

Comment:
Please rework the code style according to Maven convention in our project and use `git commit --ammend`. There should be only single commit per PR and the message should refer to JIRA ID, something like:
[SUREFIRE-12345] Support filtering of tests from Base Class (TestNG)
and you can register at our Jira here:
https://issues.apache.org/jira/projects/SUREFIRE